corehq/apps/api/urls.py

```diff
@@ -94,47 +114,63 @@ def top_level(self, request, api_name=None, **kwargs):
-    yield url(r'v0.5/odata/cases/(?P<config_id>[\w\-:]+)/\$metadata$',
-              ODataCaseMetadataView.as_view(), name=ODataCaseMetadataView.urlname)
-
-    yield url(r'v0.5/odata/forms/(?P<config_id>[\w\-:]+)/(?P<table_id>[\d]+)/$',
-              ODataFormServiceView.as_view(), name=ODataFormServiceView.table_urlname)
-    yield url(r'v0.5/odata/forms/(?P<config_id>[\w\-:]+)/$',
-              ODataFormServiceView.as_view(), name=ODataFormServiceView.urlname)
-    yield url(r'v0.5/odata/forms/(?P<config_id>[\w\-:]+)/(?P<table_id>[\d]+)/\$metadata$',
-              ODataFormMetadataView.as_view(), name=ODataFormMetadataView.table_urlname)
-    yield url(r'v0.5/odata/forms/(?P<config_id>[\w\-:]+)/\$metadata$',
-              ODataFormMetadataView.as_view(), name=ODataFormMetadataView.urlname)
-    yield url(r'v0.5/messaging-event/$', messaging_events, name="api_messaging_event_list")
-    yield url(r'v0.5/messaging-event/(?P<event_id>\d+)/$', messaging_events, name="api_messaging_event_detail")
-    yield url(r'v0\.6/case/bulk-fetch/$', case_api_bulk_fetch, name='case_api_bulk_fetch')
+urlpatterns = [
+    path('v0.5/odata/', include(odata_urlpatterns)),
+    path('odata/v1/', include(odata_urlpatterns)),
+    url(r'v0.5/messaging-event/$', messaging_events, name="api_messaging_event_list"),
+    url(r'v0.5/messaging-event/(?P<event_id>\d+)/$', messaging_events, name="api_messaging_event_detail"),
+    url(r'messaging-event/v1/$', messaging_events, name="api_messaging_event_list"),
+    url(r'messaging-event/v1/(?P<event_id>\d+)/$', messaging_events, name="api_messaging_event_detail"),
+    url(r'v0\.6/case/bulk-fetch/$', case_api_bulk_fetch),
+    url(r'case/v2/bulk-fetch/$', case_api_bulk_fetch, name='case_api_bulk_fetch'),
     # match v0.6/case/ AND v0.6/case/e0ad6c2e-514c-4c2b-85a7-da35bbeb1ff1/ trailing slash optional
-    yield url(r'v0\.6/case(?:/(?P<case_id>[\w\-,]+))?/?$', case_api, name='case_api')
-    yield from versioned_apis(API_LIST)
-    yield url(r'^case/attachment/(?P<case_id>[\w\-:]+)/(?P<attachment_id>.*)$', CaseAttachmentAPI.as_view(),
-              name="api_case_attachment")
-    yield url(r'^form/attachment/(?P<instance_id>[\w\-:]+)/(?P<attachment_id>.*)$', view_form_attachment,
-              name="api_form_attachment")
-
-    yield path('case/custom/<slug:api_id>/', generic_inbound_api, name="generic_inbound_api")
-    yield url(r'v0.5/ucr/', v0_5.get_ucr_data, name="api_get_ucr_data")
-
+    url(r'v0\.6/case(?:/(?P<case_id>[\w\-,]+))?/?$', case_api),
+    url(r'case/v2(?:/(?P<case_id>[\w\-,]+))?/?$', case_api, name='case_api'),
+    path('', include(list(versioned_apis(_OLD_API_LIST)))),
+    url(r'^case/attachment/(?P<case_id>[\w\-:]+)/(?P<attachment_id>.*)$', CaseAttachmentAPI.as_view(),
+        name="api_case_attachment"),
+    url(r'^form/attachment/(?P<instance_id>[\w\-:]+)/(?P<attachment_id>.*)$', view_form_attachment,
+        name="api_form_attachment"),
```

Comment:
Good question - I don't have a strong opinion.  I was picturing them as unversioned resources, since they just return blobs, but I suppose there's always room to grow.  What's your preference?  If we went with a v1 number, we'd need to decide which to go with:
`form/v1/attachment/instance_id/attachment_id`
`form/attachment/v1/instance_id/attachment_id`
`form_attachment/v1/instance_id/attachment_id`

The regular forms API is `form/v1`, so there's a semantic question of whether this is part of that API or an independent resource.

Reply:
My preference would be to rename them (option 3 in your list). They aren't documented anywhere currently either.